File: Sorts/BubbleSort.py
"""
Brian Taylor Vann
github.com/taylor-vann

Description:
- Module to bubble sort a list

Requirements:
- None

Methods:
- bubble_sort
"""

def bubble_sort(a):
    n = len(a)

    if n <= 1:
        return

    for i in range(n):
        for j in range(i + 1, n):
            if a[i] > a[j]:
                a[i], a[j] = a[j], a[i]

File: ECMAScript/Sorts/BubbleSort.js
// Brian Taylor Vann
// github.com/taylor-vann

const bubbleSort = (arr) => {
  if(arr.length < 2) {
    return arr;
  }

  swpd = true;

  while(swpd) {
    swpd = false;

    for(var j = 0; j < arr.length - 2; j++) {
      if(arr[j] > arr[j + 1]) {
        [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
        swpd = true;
      }
    }
  }

  return list;
};


export {
  bubbleSort
};
